app/controllers/topics_controller.rb

```diff
@@ -447,7 +447,12 @@ def status
     enabled = params[:enabled] == 'true'
 
     check_for_status_presence(:status, status)
-    @topic = Topic.find_by(id: topic_id)
+    @topic =
+      if params[:category_id]
+        Topic.find_by(id: topic_id, category_id: params[:category_id].to_i)
```

Comment:
Are we sure we want to coerce `:category_id` to `0` when it's nil?

Reply:
`params[:category_id]` can't be nil since we're checking if it's present right above ;)

Reply:
I find this odd. Why is it not enough to search a topic by its ID?